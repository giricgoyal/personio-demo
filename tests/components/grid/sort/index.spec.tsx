import { fireEvent } from '@testing-library/react'
import React from 'react'
import Sort from 'src/components/grid/sort'
import { render } from 'tests/test-utils'

describe('components/grid/sort', () => {
    let container, onChangeHandler, getByTestId

    const sortOptions = [
        {
            label: 'col1',
            value: 'col1',
        },
        {
            label: 'col2',
            value: 'col2',
        },
    ]

    beforeEach(() => {
        onChangeHandler = jest.fn()
        ;({ container, getByTestId } = render(
            <Sort sortOptions={sortOptions} sortBy="-col1" onChange={onChangeHandler} />,
        ))
    })

    test('should render correctly and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('When selection is changed', () => {
        beforeEach(() => {
            fireEvent.change(getByTestId('select'), {
                target: {
                    value: 'col2',
                },
            })
        })

        test('Should call the onChangeHandler', () => {
            expect(onChangeHandler).toHaveBeenCalledWith('-col2')
        })
    })

    describe('When sort order is changed', () => {
        beforeEach(() => {
            fireEvent.click(getByTestId('sort-order-button'))
        })

        test('should call the onChangeHandler', () => {
            expect(onChangeHandler).toHaveBeenCalledWith('col1')
        })
    })
})
