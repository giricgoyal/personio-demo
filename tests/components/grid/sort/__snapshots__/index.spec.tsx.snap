// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid/sort should render correctly and match snapshot 1`] = `
<div>
  <div>
    <i
      class="fas fa-sort "
    />
    <label
      class="select"
    >
      <span
        class="select__label"
      />
      <select
        class="select__element"
        data-testid="select"
      >
        <option
          selected=""
          value="col1"
        >
          col1
        </option>
        <option
          value="col2"
        >
          col2
        </option>
      </select>
    </label>
    <button
      class="button button--tertiary"
      data-testid="sort-order-button"
    >
      <i
        class="fas fa-arrow-down "
      />
    </button>
  </div>
</div>
`;
